router/admin.go

```diff
@@ -27,7 +35,217 @@ func (ra *RouterAdmin) Status() interface{} {
+	sqlStmt := fmt.Sprintf(`select count(*), a.error_code, a.custom_val from (select count(*), rt.job_id, st.error_code as error_code ,
+							rt.custom_val as custom_val from %[1]s rt inner join %[2]s st  on st.job_id=rt.job_id where st.job_state in 
+							('failed', 'aborted') group by rt.job_id, st.error_code, rt.custom_val) as  a group by a.custom_val, a.error_code order by a.custom_val;`, routerTableName, routerStatusTableNmae)
+	var rows *sql.Rows
+	rows, dsStats.err = dbHandle.Query(sqlStmt)
+	if dsStats.err != nil {
+		return
+	}
+	defer rows.Close()
+	result := FailedStatusErrorCodeCountsByDestination{}
+	for rows.Next() {
+		dsStats.err = rows.Scan(&result.Count, &result.ErrorCode, &result.Destination)
+		if dsStats.err != nil {
+			return
+		}
+		dsStats.FailedStatusErrorCodeCountsByDestination = append(dsStats.FailedStatusErrorCodeCountsByDestination, result)
+	}
+
+	dsStats.err = rows.Err()
+	if dsStats.err != nil {
+		return
+	}
+	dsStats.err = rows.Close()
+	return
+}
+
+func getJobCountByConnections(dbHandle *sql.DB, dsName string, dsStats *DSStats) {
+	if dsStats.err != nil {
+		return
+	}
+	routerTableName := "rt_jobs_" + dsName
+	sqlStmt := fmt.Sprintf(`select count(*), parameters->'source_id' as s, parameters -> 'destination_id' as d 
+							from %[1]s group by parameters->'source_id', parameters->'destination_id' order by parameters->'destination_id';`, routerTableName)
+	var rows *sql.Rows
+	rows, dsStats.err = dbHandle.Query(sqlStmt)
+	if dsStats.err != nil {
+		return
+	}
+	defer rows.Close()
+	result := JobCountByConnections{}
+	for rows.Next() {
+		dsStats.err = rows.Scan(&result.Count, &result.SourceId, &result.DestinationId)
+		if dsStats.err != nil {
+			return
+		}
+		dsStats.JobCountByConnections = append(dsStats.JobCountByConnections, result)
+	}
+
+	dsStats.err = rows.Err()
+	if dsStats.err != nil {
+		return
+	}
+	dsStats.err = rows.Close()
+	return
+}
+
+func getJobStatusCountsByExecTime(dbHandle *sql.DB, dsName string, dsStats *DSStats) {
+	if dsStats.err != nil {
+		return
+	}
+	routerStatusTableName := "rt_job_status_" + dsName
+	sqlStmt := fmt.Sprintf(`SELECT COUNT(*), job_state, rank FROM (SELECT job_state, RANK() OVER(PARTITION BY job_id 
+							ORDER BY exec_time DESC) as rank, job_id from %s) as inner_table GROUP BY rank, job_state order by rank, job_state`, routerStatusTableName)
+	var rows *sql.Rows
+	rows, dsStats.err = dbHandle.Query(sqlStmt)
+	if dsStats.err != nil {
+		return
+	}
+	defer rows.Close()
+	result := JobStatusCountsByExecTime{}
+	for rows.Next() {
+		dsStats.err = rows.Scan(&result.Count, &result.State, &result.Rank)
+		if dsStats.err != nil {
+			return
+		}
+		dsStats.JobStatusCountsByExecTime = append(dsStats.JobStatusCountsByExecTime, result)
+	}
+
+	dsStats.err = rows.Err()
+	if dsStats.err != nil {
+		return
+	}
+	dsStats.err = rows.Close()
+	return
+}
+
+func getUnprocessedJobCounts(dbHandle *sql.DB, dsName string, dsStats *DSStats) {
+	routerTableName := "rt_jobs_" + dsName
```

Comment:
Add jobs in the variable name to make it easy to read?
Eg. `routerJobsTable, routerJobStatusTable`

Reply:
done